File: backend/app/router/macro.py
"""宏观查询路由。"""

from fastapi import APIRouter, Depends

from app.router.dependencies import get_container
from app.router.schemas.macro import MacroIndicatorModel, MacroListModel, MacroSummaryModel
from app.router.utils import build_request, service_result_response
from app.service import ServiceContainer
from app.service.requests import MacroIndicatorRequest, MacroListRequest, MacroSummaryRequest

router = APIRouter(prefix="/api/macro", tags=["macro"])


@router.post("/indicator")
def get_macro_indicator(payload: MacroIndicatorModel, container: ServiceContainer = Depends(get_container)):
	return service_result_response(container.macro.get_macro_indicator(build_request(MacroIndicatorRequest, payload)))


@router.post("/list")
def list_macro_indicators(payload: MacroListModel, container: ServiceContainer = Depends(get_container)):
	return service_result_response(container.macro.list_macro_indicators(build_request(MacroListRequest, payload)))


@router.post("/summary")
def get_macro_summary(payload: MacroSummaryModel, container: ServiceContainer = Depends(get_container)):
	return service_result_response(container.macro.get_macro_summary(build_request(MacroSummaryRequest, payload)))


__all__ = ["router"]

File: frontend/src/api/macro.js
import request from './request'

export function getMacroIndicator(indicator_name, period = null) {
  return request.post('/api/macro/indicator', { indicator_name, period })
}

export function listMacroIndicators(indicator_names, periods = null) {
  return request.post('/api/macro/list', { indicator_names, periods })
}

export function getMacroSummary(indicator_names, recent_n = 6) {
  return request.post('/api/macro/summary', { indicator_names, recent_n })
}
